docs_app/content/guide2/writing-readable-rxjs.md

```diff
@@ -0,0 +1,251 @@
+
+Finnish Notation can be useful in situations where you may have a variable of static data that is similarly named to another variable that is an observable. For example, a component with a state property of `incomingOrders` and an observable stream of `incomingOrders$` that is continuously updating it.
+
+## Writing Effective Comments
+
+Writing effective and useful comments is the hallmark of a great engineer. The most important thing about a comment, isn't necessarily the "what", but the "why". As much as good naming and breaking up your code into custom operators may help, they can't always give the next developer all of the information they may need when they have to revisit the code.
+
+When you're thinking of the next developer, consider that they may not understand most of the abstractions you're using, let alone RxJS. And even if they do understand _all_ of the abstractions you're using, they may not understand _why_ you're using them.
+
+Let's revisit the example above with our `saveOrderAndPreventDoubleSubmission` operator and add some comments that might help the next developer understand _why_ it's written the way it is:
+
+```ts
+/**
+ * This is an operator used on an observable of save button clicks in our checkout component,
+ * when it receives a click event, it will trigger a save unless one is already in progress.
+ * If an error is encountered during save, the user will be notified. The resulting observable will
+ * only emit a value once the order has been saved successfully and the cart has been emptied.
+ */
+function saveOrderAndPreventDoubleSubmission(): OperatorFunction<any, SavedOrderResponse> {
+    // exhaustMap is used here to prevent the double submission of the saveOrder call.
+    return exhaustMap(() => {
+        orderService.saveOrder(cart).pipe(
+            // If we don't handle errors inside of the exhaustMap, save button clicks will stop working
+            // after the first error.
+            handleOrderSubmissionErrors(),
+            // It's important that we empty our cart inside of the exhaustMap, to ensure
+            // that work is done before another save order is triggered. Otherwise, a user
+            // may end up trying to save the same cart data twice.
+            emptyCartAfterSave(),
+        )
+    });
+}
+```
+
+What the goal is for the comments above is really just to give the next developer (or your future self) more context as to why you made particular decisions with the code. Some of it may be obvious to you now, or it may become more obvious to you over time, but that doesn't mean you should not create useful comments for people that don't have the same level of understanding of the code base as you. The "What if you get hit by a bus?" factor should always apply.
```

Comment:
Consider changing the first sentence to "The goal for the comments above is ..." to prevent the double "is".